.github/actions/check-submodule-reference/action.yml

```diff
@@ -0,0 +1,25 @@
+name: "Check submodule reference"
+description: "Check that submodule references a commit in a submodule repository's branch"
+
+inputs:
+  submodule_path:
+    description: "directory with the submodule"
+    required: true
+  branch:
+    description: "branch in the submodule repository"
+    required: true
+
+runs:
+  using: "composite"
+
+  steps:
+    - name: Check ${{ inputs.submodule_path }} submodule reference
+      shell: bash -euxo pipefail {0}
+      run: |
+        cd ${{ inputs.submodule_path }}
+        git fetch origin "refs/heads/${{ inputs.branch }}:refs/heads/origin/${{ inputs.branch }}"
```

Comment:
I would change these strings to be wrapped in single quotes rather than double quotes to avoid some sort of shell injection attack.